py3status/modules/mpd_status.py

```diff
@@ -110,12 +110,30 @@ class Py3status:
     state_play = '[play]'
     state_stop = '[stop]'
 
+    def _connection(self):
+        try:
+            if self.connection is None:
+                self.connection = MPDClient()
+                self.connection.connect(host=self.host, port=self.port)
+                if self.password:
+                    self.connection.password(self.password)
+            return self.connection
+        except (socket.error, ConnectionError, CommandError) as e:
+            self.connection = None
+            raise e
+
+    def _mpd_disconnect(self):
+        self._connection().disconnect()
+        self.connection = None
+
     def post_config_hook(self):
```

Comment:
Keep `post_config_hook` on top for consistency.

Reply:
sure! ;-)